presto-hive/src/main/java/com/facebook/presto/hive/HiveConnectorFactory.java

```diff
@@ -135,6 +137,7 @@ public Connector create(String catalogName, Map<String, String> config, Connecto
             HiveAnalyzeProperties hiveAnalyzeProperties = injector.getInstance(HiveAnalyzeProperties.class);
             ConnectorAccessControl accessControl = new PartitionsAwareAccessControl(injector.getInstance(ConnectorAccessControl.class));
             Set<Procedure> procedures = injector.getInstance(Key.get(new TypeLiteral<Set<Procedure>>() {}));
+            ConnectorPlanOptimizerProvider planOptimizerProvider = injector.getInstance(HivePlanOptimizerProvider.class);
```

Comment:
- There are test failures. `HivePlanOptimizerProvider` needs to be bound in `HiveClientModule`:
`binder.bind(ConnectorPlanOptimizerProvider.class).to(HivePlanOptimizerProvider.class).in(Scopes.SINGLETON);`
- Also the injector needs to get `ConnectorPlanOptimizerProvider`.

Reply:
Good catch! 

That was a quick review. I was still testing locally :)